hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -156,18 +156,18 @@ withDirectChain tracer networkMagic iocp socketPath keyPair party cardanoKeys ca
                 { postTx = \tx -> do
                     traceWith tracer $ ToPost tx
 
-                    res <- timeout 10 $
-                      atomically $ do
-                        fromPostChainTx wallet headState cardanoKeys tx >>= \case
-                          Nothing ->
-                            pure Nothing
-                          Just partialTx ->
-                            Just <$> finalizeTx wallet headState partialTx
-
-                    maybe
-                      (callback PostTxFailed)
-                      (atomically . writeTQueue queue)
-                      (join res)
+                    ( do
+                        res <-
+                          timeout 10 $
+                            atomically $
+                              fromPostChainTx wallet headState cardanoKeys tx
+                                >>= finalizeTx wallet headState
+                                >>= writeTQueue queue
+                        when (isNothing res) $ throwIO (NoSeedInput @CardanoTx)
+                      )
+                      `catch` \(e :: InvalidTxError CardanoTx) -> do
+                        traceWith tracer $ PostingTxFailed tx e
+                        throwIO e
```

Comment:
Rather than catching, logging & rethrowing here, I'd expect that to happen where `postTx` is used, e.g. to not have the thread / process crash. Which brings me to the point: What should happen in these cases, should our "HeadLogic" be able to react on this situation? Shall it produce a server output for the clients? I guess that catching and converting it to an `Event` would be necessary for that?